src/api/search.api.js

```diff
@@ -272,6 +281,141 @@ async function searchLayerFeatures(outputProjection, queryString, layer, lang, c
+    return searchFields.some((field) => {
+        const value = feature.values_[field]?.toString()
+        return !!value && value.trim().toLowerCase().includes(queryStringClean)
+    })
+}
+
+/**
+ * Searches for the query string in the KML layer
+ *
+ * @param {CoordinateSystem} outputProjection
+ * @param {String} queryString
+ * @param layer
+ * @returns {SearchResult[]}
+ */
+async function searchKmlLayerFeatures(outputProjection, queryString, layer) {
+    try {
+        const searchFields = ['name', 'description', 'id']
+        const features = parseKml(layer, outputProjection, [])
+        if (!features || !features.length) {
+            return []
+        }
+
+        const includedFeatures = features.filter((feature) =>
+            isQueryInFeature(feature, queryString, searchFields)
+        )
+        if (!includedFeatures.length) {
+            return []
+        }
+
+        return includedFeatures.map((feature) =>
+            createSearchResultFromLayer(layer, feature, outputProjection)
+        )
+    } catch (error) {
+        log.error(
+            `Failed to search layer features for layer ${layer.id}, fallback to empty result`,
+            error
+        )
+        return []
+    }
+}
+
+/**
+ * Searches for the query string in the GPX layer
+ *
+ * @param {CoordinateSystem} outputProjection
+ * @param {String} queryString
+ * @param layer
+ * @returns {SearchResult[]}
+ */
+async function searchGpxLayerFeatures(outputProjection, queryString, layer) {
+    try {
+        const searchFields = ['name', 'description', 'id']
+        const features = parseGpx(layer.gpxData, outputProjection, [])
+        if (!features || !features.length) {
+            return []
+        }
+        const includedFeatures = features.filter((feature) =>
+            isQueryInFeature(feature, queryString, searchFields)
+        )
+        if (!includedFeatures.length) {
+            return []
+        }
+        return includedFeatures.map((feature) =>
+            createSearchResultFromLayer(layer, feature, outputProjection)
+        )
+    } catch (error) {
+        log.error(
+            `Failed to search layer features for layer ${layer.id}, fallback to empty result`,
+            error
+        )
+        return []
+    }
+}
+
+/**
+ * Creates the SearchResult for a layer
+ *
+ * @param layer
+ * @param feature
+ * @param {ol/extent|null} extent
+ * @param {CoordinateSystem} outputProjection
+ * @returns {SearchResult}
+ */
+function createSearchResultFromLayer(layer, feature, outputProjection) {
+    const featureName = feature.values_.name || layer.name || ''
+    const coordinates = extractOlFeatureCoordinates(feature)
+    const zoom = outputProjection.get1_25000ZoomLevel()
+
+    const coordinatePoints = points(coordinates)
+    const centerPoint = center(coordinatePoints)
+    const normalExtent = normalizeExtent(bbox(coordinatePoints))
+
+    const featureId = feature.id_ ?? feature.ol_uid
```

Comment:
I think you could also try to run `feature.get('id')` (it might be cleaner than using private `feature.id_`). But that presuppose `feature` is of type `ol/Feature`